src/thunderbots/software/geom/util.cpp

```diff
@@ -752,6 +752,59 @@ std::optional<Point> lineIntersection(const Vector &a, const Vector &b, const Ve
     return std::make_optional(intersection);
 }
 
+std::optional<Point> finiteLengthLineIntersection(const Vector &a, const Vector &b,
+                                                  const Vector &c, const Vector &d)
+{
+    std::optional<Point> intersection = lineIntersection(a, b, c, d);
+
+    if (intersection != std::nullopt)
+    {
+        // We know the point returned by lineIntersection() exists along the trajectory of
+        // one of the lines, now we must check that it is within the specified line points
+        // If point C exists on line A-B, DIST(AC) + DIST(CB) = DIST(AB)
+        // Calculated for lines a-b and c-d
+        if ((fabs(dist(a, *intersection)) + fabs(dist(*intersection, b)) ==
+             fabs(dist(a, b))) &&
+            fabs(dist(c, *intersection)) + fabs(dist(*intersection, d)) ==
+                fabs(dist(c, d)))
+        {
+            return intersection;
+        }
+    }
+
+    return std::nullopt;
+}
+
+std::optional<Point> velocity_line_intersection(const Vector element_velocity,
+                                                const Vector element_position,
+                                                const Point line_start,
+                                                const Point line_end,
+                                                const Rectangle boundry_area)
+{
+    // If the element is not within the boundries, return null
+    if (!boundry_area.containsPoint(element_position))
+    {
+        return std::nullopt;
+    }
+
+    // Calculate the closest edge of the boundry based on element speed and position
```

Comment:
Why does this need to be based on time? Isn't this just the intersection of the line formed by the velocity and the rectangle? Seems like we don't need the concept of units/time here (it's up to the caller to "know" what it's doing).

(same question for the `seconds_to_closest_edge` stuff below)

Reply:
I could definitely use line intersection functions here, but this seemed to me like the simplest way to find where the velocity will intersect the lines of the boundary rectangle. There exists a line-intersect-rectangle function but it considers all intersections as if the velocity line is infinite.

I've changed references to 'seconds' to 'proximity' as a catch-all term.

Reply:
Changes functions (as suggested by Gareth), no longer applicable.